server/lib/chat/chat-service.ts

```diff
@@ -565,6 +567,49 @@ export default class ChatService {
     }
   }
 
+  async searchChannels({
+    userId,
+    limit,
+    offset,
+    searchStr,
+  }: {
+    userId: SbUserId
+    limit: number
+    offset: number
+    searchStr?: string
+  }): Promise<SearchChannelsResponse> {
+    const { channels, joinedChannels } = await searchChannelsAsUser({
+      userId,
+      limit,
+      offset,
+      searchStr,
+    })
+
+    const userJoinedChannelsSet = new global.Set(joinedChannels)
+
+    const [detailedChannelInfos, joinedChannelInfos] = channels.reduce<
```

Comment:
Can you just write this as a for loop please? This is an extremely complex way of doing "iterate through a list and occasionally push values into 2 lists"